pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -192,38 +190,32 @@ func (r *ReconcileNSTemplateTier) Reconcile(request reconcile.Request) (reconcil
 // the templateRefs of the given NSTemplateTier.
 //
 // (longer explanation)
-// newLabelSelector creates a selector to find MasterUserRecords with the following labels:
-// - toolchain.dev.openshift.com/<member>-templates-tier: <tier>
-// - toolchain.dev.openshift.com/<member>-templates-tier-hash: !<hash>
+// newLabelSelector creates a selector to find MasterUserRecords which have a label with key
+// `toolchain.dev.openshift.com/<tiername>-tier-hash` but whose value is NOT `<hash>`
 //
 // In other words, this label selector will be used to list MasterUserRecords which have a user account set to the given `<tier>`
 // but with a template version (defined by `<hash>`) which is NOT to the expected value (the one provided by `instance`).
 //
 // Note: The `hash` value is computed from the TemplateRefs. See `computeTemplateRefsHash()`
-func newLabelSelector(instance *toolchainv1alpha1.NSTemplateTier, member cluster.FedCluster) (client.MatchingLabelsSelector, error) {
+func newLabelSelector(tier *toolchainv1alpha1.NSTemplateTier) (client.MatchingLabelsSelector, error) {
 	// compute the hash of the `.spec.namespaces[].templateRef` + `.spec.clusteResource.TemplateRef`
-	hash, err := ComputeTemplateRefsHash(instance)
+	hash, err := ComputeHashForNSTemplateTier(*tier)
 	if err != nil {
 		return client.MatchingLabelsSelector{}, err
 	}
 	selector := labels.NewSelector()
-	tierLabel, err := labels.NewRequirement(TemplateTierNameLabel(member.Name), selection.Equals, []string{instance.Name})
+	tierLabel, err := labels.NewRequirement(TemplateTierHashLabelKey(tier.Name), selection.Exists, []string{})
 	selector = selector.Add(*tierLabel)
-	templateHashLabel, err := labels.NewRequirement(TemplateTierHashLabel(member.Name), selection.NotEquals, []string{hash})
+	templateHashLabel, err := labels.NewRequirement(TemplateTierHashLabelKey(tier.Name), selection.NotEquals, []string{hash})
```

Comment:
Hm.. I thought that this selector would be enough and you don't need to use `selection.Exists` but maybe you are right and without `Exists` it will return resources without that label too.